### PR TITLE
Feature/cordova app auto exit

### PR DESCRIPTION
**Pull request by shiljo-litmus7**
Issues Fixed
New Cordova Android plugin to close the app programatically. The plugin will subscribe to message with type EXIT_CLIENT_APP and will trigger the app exit using the Cordova Plugin
This capability is added as part of the implementation of https://petcoalm.atlassian.net/browse/PDPOS-6897

Summary
 A new OpenPos Cordova Platform plugin for closing Android app

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    name(): string {
+        return 'ExitApp';
+    }
+
+    pluginPresent(): boolean {
+        return !!window.hasOwnProperty('ExitApp');
+    }
+
+    initialize(): Observable<string> {
+        return Observable.create((initialized: Subject<string>) => {
```
Comment (jrmihalick) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts:27]: TS lint usually warns me not to use `Observable.create` and now use new `Observable<string>`.  Not sure why its deprecated.  https://stackoverflow.com/questions/55539103/angular-create-is-deprecated-use-new-observable-instead

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            // tslint:disable-next-line:no-string-literal
+            this.ExitAppPlugin = window['ExitApp'];
+            if (!this.ExitAppPlugin) {
+                initialized.error(`Tried to initialize plugin ${this.name()} which is not present`);
+            }
+
+            this.sessionService.getMessages('Proxy').pipe(
```
Comment (jrmihalick) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts:34]: We don't need to use Proxy messages here.  You can search on this in the client project and find a lot of examples...  `sessionService.getMessages(`

File: openpos-server-core/src/main/java/org/jumpmind/pos/server/model/ProxyTypeConstants.java [hunk 3]
```diff
@@ -3,5 +3,6 @@
 public class ProxyTypeConstants {
     public static final String PAYMENT = "Payment";
     public static final String PRINT = "Print";
+    public static final String APP_MANAGEMENT = "AppManagement";
```
Comment (jrmihalick) [inline openpos-server-core/src/main/java/org/jumpmind/pos/server/model/ProxyTypeConstants.java:6]: See prior comments, we don't need to use a Proxy message here.

**Review by jrmihalick — CHANGES_REQUESTED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts (hunk 1 above)
Comment (shiljo-litmus7) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts]: Done

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts (hunk 2 above)
Comment (shiljo-litmus7) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/cordova-plugins/auto-exit.ts]: Done

File: openpos-server-core/src/main/java/org/jumpmind/pos/server/model/ProxyTypeConstants.java (hunk 3 above)
Comment (shiljo-litmus7) [reply openpos-server-core/src/main/java/org/jumpmind/pos/server/model/ProxyTypeConstants.java]: Done

**Review by jrmihalick — APPROVED** [review]

**Pull request merged**